AdobeStockAsset/Model/Client.php

```diff
@@ -32,32 +32,35 @@ public function __construct(ConfigInterface $config)
     }
 
     /**
-     * @return mixed
+     * @param \Magento\AdobeStockAssetApi\Api\Data\RequestInterface $request
+     * @return array
+     * @throws \AdobeStock\Api\Exception\StockApi
      */
-    public function search()
+    public function execute(\Magento\AdobeStockAssetApi\Api\Data\RequestInterface $request)
     {
-        // TODO THIS IS A STUB
-        $results_columns = Constants::getResultColumns();
-        $search_params = new SearchParameters();
-        $search_params->setWords('tree')->setLimit(32)->setOffset(0);
+        // TODO: THIS IS A STUB. SHOULD BE REFACTORED
+        $words = $request->getData('filters')['words'] ?? 'image';
 
-        $result_column_array = [
-            $results_columns['COMP_URL'],
-            $results_columns['ID'],
-            $results_columns['NB_RESULTS'],
-        ];
+        $searchParams = new SearchParameters();
+        $searchParams->setWords($words);
+        $searchParams->setLimit($request->getData('size'));
+        $searchParams->setOffset($request->getData('offset'));
+
+        $resultsColumns = Constants::getResultColumns();
+        $resultColumnArray = [];
+        foreach ($request->getData('resultColumns') as $column) {
+            $resultColumnArray[] = $resultsColumns[$column['field']];
+        }
 
         $request = new SearchFilesRequest();
         $request->setLocale('En_US');
```

Comment:
I think locale should be passed through the `RequestInterface`. (let's add a TODO if we'd like to extract it later)